appinventor/blocklyeditor/src/backpack.js

```diff
@@ -380,14 +380,11 @@ Blockly.Backpack.prototype.removeFromBackpack = function(ids) {
   var p = this;
   this.getContents(function(/** @type {string[]} */ contents) {
     if (contents && contents.length) {
-      for (var i = 0; i < contents.length; i++) {
-        var xml = Blockly.Xml.textToDom(contents[i]);
-        var blockID = xml.firstElementChild.getAttribute("id");
-        if (ids.indexOf(blockID) >= 0) {
-          contents.splice(i, 1);
-          i--;
-        }
-      }
+      var blockInBackPack = p.workspace_.backpack_.flyout_.workspace_.topBlocks_.map(function(elt) {
+        return elt.id;
+      });
+      var index = blockInBackPack.indexOf(ids[0]);
+      contents.splice(index, 1);
```

Comment:
I don't think this would happen, but in theory there is a possibility that ids[0] doesn't exist in the blockInBackPack array. You should check to see if `index > -1` before trying to splice it out of the array.